pkg/logql/log/pipeline.go

```diff
@@ -381,3 +391,32 @@ func unsafeGetBytes(s string) []byte {
 func unsafeGetString(buf []byte) string {
 	return *((*string)(unsafe.Pointer(&buf)))
 }
+
+func replaceChars(str string, offsets []int) string {
+	offsets = offsets[:0]
+	for i, r := range str {
```

Comment:
looking at prometheus [NormalizeLabel](https://github.com/grafana/loki/blob/cdafa90e62862c69b46f5d6b592caf9cfc87f287/vendor/github.com/prometheus/prometheus/storage/remote/otlptranslator/prometheus/normalize_label.go#L42), we also need to handling labels starting with a number

Reply:
great find. I didn't realize `NormalizeLabel` was a thing!